src/logger/mongoose_json_formatter.erl

```diff
@@ -11,24 +11,63 @@
 -export([format/2]).
 
 -spec format(logger:log_event(), logger:formatter_config()) -> unicode:chardata().
-format(E = #{msg := {report, #{label := {error_logger, _}, format := Format, args := Terms}}}, FConfig) ->
-    format(E#{msg := {report,
+format(Event, FConfig) ->
+    try do_format(Event, FConfig)
+    catch
+        %% Exceptions during log formatting can lead to a death spiral of recursive error logging, so
+        %% format the formatter error in a safe way and don't allow the exception to propagate.
+        Class:Reason -> format_log_formatter_error(Class, Reason, Event, FConfig)
+    end.
+
+
+format_log_formatter_error(Class, Reason, #{meta := #{time := Time, mfa := Mfa, file := File, line := Line}}, FConfig) ->
```

Comment:
We have to include the complete Event, encoded as a term.

`original_event => unicode:characters_to_binary(io_lib:format("~0p", Event))`

Reply:
I've added this and pushed new changes